src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshServer.cs

```diff
@@ -38,18 +38,25 @@ public BrowserRefreshServer(IReporter reporter)
             _taskCompletionSource = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
         }
 
-        public async ValueTask<string> StartAsync(CancellationToken cancellationToken)
+        public async ValueTask<IEnumerable<string>> StartAsync(CancellationToken cancellationToken)
         {
             var envHostName = Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME");
             var hostName = envHostName ?? "127.0.0.1";
 
-            var useTls = await ShouldUseHttps();
+            var supportsTLS = await SupportsTLS();
 
             _refreshServer = new HostBuilder()
                 .ConfigureWebHost(builder =>
                 {
                     builder.UseKestrel();
-                    builder.UseUrls(useTls ? $"https://{hostName}:0" : $"http://{hostName}:0");
+                    if (supportsTLS)
```

Comment:
Do we need to bother with this check if we can fallback on the logic that makes the WebSocket connection to the valid URL on the client-side?

Feels like we can avoid making the additional process call here.

Reply:
Kestrel does not like it if you ask it to bind to a TLS port without the certs installed. So we do need to do some due-diligence here.